selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceProjectsSamples.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.pageobject.dashboard.workspaces;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.BottomButton.ADD_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.BottomButton.CANCEL_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.Locators.CHECKBOX_BY_SAMPLE_NAME_ID_TEMPLATE;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.Locators.SAMPLE_ITEM_TEMPLATE_XPATH;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.Locators.TAB_STATE_TEMPLATE_XPATH;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.UpperButton.BLANK_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.UpperButton.GITHUB_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.UpperButton.GIT_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.UpperButton.SAMPLES_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.UpperButton.ZIP_BUTTON;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.util.List;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+@Singleton
+public class WorkspaceProjectsSamples {
+
+  private final SeleniumWebDriverHelper seleniumWebDriverHelper;
+
+  @Inject
+  private WorkspaceProjectsSamples(SeleniumWebDriverHelper seleniumWebDriverHelper) {
+    this.seleniumWebDriverHelper = seleniumWebDriverHelper;
+  }
+
+  public interface Locators {
+    String SAMPLE_ITEM_TEMPLATE_XPATH =
+        "//ng-transclude[@class='che-list-item-content']//span[text()='%s']";
+    String TAB_STATE_TEMPLATE_XPATH = "//che-toggle-joined-button[@id='%s']/button";
+    String CHECKBOX_BY_SAMPLE_NAME_ID_TEMPLATE = "sample-%s";
+  }
+
+  private interface Button {
+    String get();
+  }
+
+  public enum UpperButton implements Button {
```

Comment:
Position of buttons could be changed later, so it would be better not to use position in enum name to avoid misleading. What about `ActionButton` name instead of `UpperButton`?